src/components/ClipBoardButton.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+import { FileCopy } from "@mui/icons-material";
+import Tooltip from "@mui/material/Tooltip";
+import IconButton from "@mui/material/IconButton";
+
+const ClipboardButton = ({ textToCopy }) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const handleCopyToClipboard = (value) => {
+    navigator.clipboard.writeText(value);
+    setIsCopied(true);
+    setTimeout(() => {
+      setIsCopied(false);
+    }, 2000);
+  };
+
+  return (
+    <Tooltip
+      title={isCopied ? "Copied" : "Copy to clipboard"}
+      placement="bottom"
+    >
+      <IconButton onClick={() => handleCopyToClipboard(textToCopy)}>
```

Comment:
There's no need to pass textToCopy in here since it is not dependent on the click callback. you could do `onClick={handleCopyToClipboard}`, and reference textToCopy inside the handler function where it would be available as it is a closure.